### PR TITLE
Refactor some class components to functional

### PR DESCRIPTION
**Pull request by PavelJankoski**
#### Summary

References #6376 

#### Changes

List of refactored components:

- /pkg/webui/components/form/field/stories/shared.js
- /pkg/webui/components/input/stories/shared.js
- /pkg/webui/components/profile-dropdown/index.js
- /pkg/webui/components/breadcrumbs/context.js
- /pkg/webui/components/tag/group/index.js
- /pkg/webui/lib/components/date-time/index.js
- /pkg/webui/lib/components/animation/index.js
- /pkg/webui/lib/components/env.js
- /pkg/webui/lib/errors/custom-errors.js -> **not a component**
- /pkg/webui/lib/yup.js -> **not a component**
- /pkg/webui/console/components/webhook-template-form/index.js
- /pkg/webui/console/components/webhook-form/index.js
- /pkg/webui/console/components/pubsub-form/index.js
- /pkg/webui/console/components/events/error-boundary.js
- /pkg/webui/console/components/location-form/index.js
-  /pkg/webui/console/components/join-eui-prefixes-input/index.js - **removed**
- /pkg/webui/console/containers/gateway-connection/gateway-connection.js

#### Notes for Reviewers

When testing, please pay more attention to the refactored form components because they are large and you know the business logic more briefly(if everything behaves as expected).

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] The steps/process to test this feature are clearly explained including testing for regressions.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/webui/console/components/location-form/index.js [hunk 1]
```diff
@@ -89,295 +88,294 @@ const emptyLocation = {
...
+          () => {
+            setMapCenter(defaultLocation)
+            setZoom(2)
+          },
+        )
       }
     }
-    this.setState(newState)
-  }
+    setLoading(false)
```
Comment (kschiffer) [inline pkg/webui/console/components/location-form/index.js]: The loading should also be set to false within the callback above.

File: pkg/webui/console/components/location-form/index.js [hunk 2]
```diff
@@ -89,295 +88,294 @@ const emptyLocation = {
...
-            step="any"
-            title={sharedMessages.longitude}
-            description={sharedMessages.longitudeDesc}
-            name="longitude"
-            component={Input}
-            required={!updatesDisabled}
-            disabled={updatesDisabled}
-            onBlur={this.handleLongitudeChange}
+  return (
+    <React.Fragment>
```
Comment (kschiffer) [inline pkg/webui/console/components/location-form/index.js]: Looks like this doesn't actually need to be wrapped by a React.Fragment.

File: pkg/webui/console/views/application-integrations-lora-cloud/index.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ import LoRaCloudImage from '@assets/misc/lora-cloud.png'
 
 import PageTitle from '@ttn-lw/components/page-title'
 import Breadcrumb from '@ttn-lw/components/breadcrumbs/breadcrumb'
-import { useBreadcrumbs, withBreadcrumb } from '@ttn-lw/components/breadcrumbs/context'
+import { useBreadcrumbs } from '@ttn-lw/components/breadcrumbs/context'
```
Comment (kschiffer) [inline pkg/webui/console/views/application-integrations-lora-cloud/index.js:26]: You're not using `useBreadcrumbs` here.

File: pkg/webui/lib/components/env.js
Comment (kschiffer) [inline pkg/webui/lib/components/env.js:1]: This whole module can actually be removed entirely (and remaining usage refactored). We're now using the environment selectors directly via `/lib/selectors/env.js`.

**Review by kschiffer — CHANGES_REQUESTED** [review]
Great work and a lot of changes indeed. Let's see what the e2es say. 
I've found some minor issues.

**Comment by kschiffer** [comment]
@ryaplots please take a look as well

File: pkg/webui/console/views/application-integrations-lora-cloud/index.js (hunk 3 above)
Comment (PavelJankoski) [reply pkg/webui/console/views/application-integrations-lora-cloud/index.js:26]: It is used from the previous implementation. The HOC was redundant.
```
useBreadcrumbs(
    'apps.single.integrations.lora-cloud',
    <Breadcrumb
      path={`/applications/${appId}/integrations/lora-cloud`}
      content={sharedMessages.loraCloud}
    />,
  )
```

File: pkg/webui/console/components/location-form/index.js [hunk 4]
```diff
@@ -89,295 +88,293 @@ const emptyLocation = {
...
       }
     }
-    this.setState(newState)
-  }
-
-  @bind
-  async handleSubmit(values, { setSubmitting }) {
-    const { onSubmit, entityId, validationSchema } = this.props
-    this.setState({ error: undefined })
+  }, [additionalMarkers.length, initialValues])
```
Comment (ryaplots) [inline pkg/webui/console/components/location-form/index.js:151]: In this bit, in the original implementation there is a "default" setState you did not include here (`this.setState(newState)`), why?

**Review by ryaplots — CHANGES_REQUESTED** [review]
Nice work. Just a small issue with the pubsub form according to the e2e tests.

See my remark.

File: pkg/webui/console/components/location-form/index.js (hunk 4 above)
Comment (PavelJankoski) [reply pkg/webui/console/components/location-form/index.js:151]: You are right, I re-do the logic according to the old component

**Review by kschiffer — APPROVED** [review]
Thank you. Please clean up the commit trail before merging. Any fix commits should be squashed into their respective earlier commit. You can use interactive rebasing to achieve that.

**Review by ryaplots — APPROVED** [review]

**Pull request merged**